src/shell.js

```diff
@@ -133,16 +133,24 @@ var ENVIRONMENT_IS_WASM_WORKER = Module['$ww'];
 #if SHARED_MEMORY && !MODULARIZE
 // In MODULARIZE mode _scriptDir needs to be captured already at the very top of the page immediately when the page is parsed, so it is generated there
 // before the page load. In non-MODULARIZE modes generate it here.
-var _scriptDir = (typeof document != 'undefined' && document.currentScript) ? document.currentScript.src : undefined;
+var _scriptDir;
 
-if (ENVIRONMENT_IS_WORKER) {
-  _scriptDir = self.location.href;
-}
 #if ENVIRONMENT_MAY_BE_NODE
-else if (ENVIRONMENT_IS_NODE) {
+if (ENVIRONMENT_IS_NODE) {
   _scriptDir = __filename;
-}
+} else
 #endif // ENVIRONMENT_MAY_BE_NODE
+#if ENVIRONMENT_MAY_BE_WORKER
+if (ENVIRONMENT_IS_WORKER) {
+  _scriptDir = self.location.href;
+} else
+#endif // ENVIRONMENT_MAY_BE_WORKER
+#if ENVIRONMENT_MAY_BE_WEB
+if (typeof document != 'undefined' && document.currentScript) { // web
+  _scriptDir = document.currentScript.src;
+} else
+#endif // ENVIRONMENT_MAY_BE_WEB
+  _scriptDir = undefined;
```

Comment:
Is explicit assignment here necessary / would it be not `undefined` already?

Reply:
Indeed, it isn't necessary. Perhaps there's some sore of "no-op" statement in JS that I could write in this `} else`-block?

Reply:
Hm can you just remove `else`? If not, the empty statement (literally just `;`) can work. That might look like a code smell though, so feel free to replace with empty block instead (`{}`).

Reply:
Indeed, this is not ideal. I'll think about the best solution. FWIW, I considered doing `assert(0);` (or a JS function similar to `__builtin_trap();` / `__builtin_unreachable();`) here, but that would probably cause issues with `ENVIRONMENT_IS_SHELL` / `-sENVIRONMENT=shell` hence the mention of "no-op".

Perhaps this should fallback to `_scriptDir = './';` instead? Though, that would probably cause issues with this: 
https://github.com/emscripten-core/emscripten/blob/3b124a624fb15782a0eba452377969d89bd57cef/src/shell.js#L380-L384
(I still don't quite understand the need of `scriptDirectory`, hence the TODO-item added in commit 3e47701f8b58fec13a1b61c13f0b3fad4af833c4)